### PR TITLE
nsyshid: Experimental libusb build

### PR DESCRIPTION
**Pull request by deReeperJosh**
Just creating this for visibility - and for any interested parties who may know more than I do. I am trying to make the nsyshid library cross platform, so that it can be used on MacOS and Linux. This is step 1 in trying to port some code that I have worked on to bring Emulated USB devices such as the Skylanders Portal/Infinity Base/Dimensions Toypad. LibUSB requires some more information in order to actually be able to send such requests though, such as bmRequest/bmRequestType, endpoint and some other bits of data. If anyone has some ideas about where to obtain this info it would be much appreciated! 

I would also like to note that I haven't implemented libusb prior to attempting this - and I am currently in the process of moving overseas so haven't got access to my Wii U at the time of opening this PR, so this may take a while.

### CODE REVIEW

**Comment by TheRealWario** [comment]
Does this bear any relevance to my post? https://github.com/cemu-project/Cemu/issues/902

**Comment by deReeperJosh** [comment]
It doesn't, but other libusb implementations (RPCS3 and Dolphin) have working Trap Team audio transfers, so the speakers should work on Cemu after this :)

File: src/Cafe/CMakeLists.txt [hunk 1]
```diff
@@ -485,7 +485,10 @@ endif()
 
 set_property(TARGET CemuCafe PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
-target_include_directories(CemuCafe PUBLIC "../")
+target_include_directories(CemuCafe PUBLIC 
+  "../" 
+  ${LIBUSB_INCLUDE_DIR}
+)
```
Comment (Tachi107) [inline src/Cafe/CMakeLists.txt:491]: Please avoid manually adding a list of include dirs and libraries like these, and use imported targets instead. I'm not sure how to patch vcpkg dependencies to add imported targets when they are not already provided, but you could always submit a patch to <https://github.com/microsoft/vcpkg/tree/master/ports/libusb>.

Comment (deReeperJosh) [reply src/Cafe/CMakeLists.txt:491]: I was just going off of the usage for the libusb vcpkg, not sure if I would actually know what needs to change to submit a package there either but I will look in to it

**Comment by LuismaSP89** [comment]
Would this libusb eventually bring us the possibility to use real wiimote support on the emulator? Just like the bluetooth passthrough on dolphin, were you can use the original bluetooth controller of the Wii instead of emulating it.

**Comment by deReeperJosh** [comment]
In theory yes, but I wouldn't be able to implement that functionality myself, not yet anyway

**Comment by ssievert42** [comment]
Hey, I opened #946 yesterday (have been working on and off on this and had it kind of working since may 2023), but totally missed this PR :sweat_smile:
Completely emulated devices sound like a great idea!
In #946, I've got the real Lego Dimensions portal to work reliably on linux :)
An unsolved issue are still controls transfers, but at least the Lego Dimensions portal seems to not really need them.
In theory, my implementation should be cross platform, but I've got no way to test on Windows or macOS.

My suggestion would be, that I'd restructure #946 a bit to allow implementing virtual devices, and once that is merged, you could then use that as a base for fancy emulated devices.
Do you think that requiring a device to have the following functionality would be enough?
- get vendor and product id
- get max packet sizes for RX and TX
- try to write a block of data to the device and return the bytes written
- try to read a block of data into a buffer and return the bytes read
- get descriptors as implemented in `export_HIDGetDescriptor`
- try to do control transfers with some of the arguments of `_hidSetReportSync`

**Comment by deReeperJosh** [comment]
Oh that's great! I actually haven't been able to touch this work since I opened the PR, so honestly just do whatever works, I think creating some sort of Device class, that can be extended to be either a USB device or an Emulated Device would be ideal, but I will leave that up to you! And I will close off this PR, and collaborate with you as needed on your PR

**Pull request closed without merging**